strutil/shlex/shlex.go

```diff
@@ -414,3 +415,28 @@ func Split(s string) ([]string, error) {
 		subStrings = append(subStrings, word)
 	}
 }
+
+// Join returns a shell-escaped string version of the slice s
+func Join(s []string) string {
+	quoted := make([]string, len(s))
+	for i := range s {
+		quoted[i] = Quote(s[i])
+	}
+	return strings.Join(quoted, " ")
+}
+
+var unsafeQuoteChars = regexp.MustCompile(`[^\w@%+=:,./-]`)
+
+// Quote returns a shell-escaped version of the string s
+func Quote(s string) string {
+	if s == "" {
+		return `''`
+	}
+	if unsafeQuoteChars.Find([]byte(s)) == nil {
```

Comment:
Probably better to use `.FindString(s)` and compare `== ""` to avoid the allocation/conversion to `[]byte`. (The `""` case has been checked above so doesn't matter here.)

Reply:
Thanks. Done.